File: go/p7.go
package main

import (
	"math"
)

func isPrime(n int) bool {
	if n <= 1 {
		return false
	}
	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func p7() []int {
	primes := []int{}
	isPrimeCounter := 0
	for i := 2; ; i++ {
		if isPrime(i) {
			primes = append(primes, i)
			isPrimeCounter++
		} else if isPrimeCounter >= 10001 {
			break
		}
	}

	return primes
}

File: cpp/p7.cpp
#include <iostream>

bool isPrime(int num)
{
    if (num <= 1)
    {
        return false;
    }

    for (int i = 2; i * i <= num; ++i)
    {
        if (num % i == 0)
        {
            return false;
        }
    }

    return true;
}

int main(int argc, char const *argv[])
{
    int isPrimeCounter = 0;
    int num = 0;
    int ans = 0;

    while (isPrimeCounter <= 10001)
    {
        num++;

        if (isPrime(num))
        {
            isPrimeCounter++;
        }
        
        if (isPrimeCounter == 10001)
        {
            ans = num;
            break;
        }
    }

    std::cout << ans << std::endl;
    
    return 0;
}
